### PR TITLE
Re-architect SlackKit/Vapor interplay

### PR DESCRIPTION
**Pull request by allenhumphreys**
Since we're using the webAPI to send messages out instead of the RTM client, I set everything up so that each container gets it's own outgoing "client". There is now a singleton RTM `SlackListener` that services get registered with. The service and container are registered together so that we safely use them on the correct event loop. I also plan to rotate which one is chosen so as to avoid blocking one continuously.

I think this is a lot cleaner, if not entirely perfect.

### CODE REVIEW

File: Sources/App/Controllers/OnTapController.swift [hunk 1]
```diff
@@ -112,16 +112,23 @@ func set(beer newBeer: Beer?, on tap: Tap) -> Bool {
     return false
 }
 
-final class OnTapController: RouteCollection, ServiceType {
+extension OnTapController: ServiceType {
```
[the remark this answers is not in the document]
Comment (allenhumphreys) [reply Sources/App/Controllers/OnTapController.swift]: Yeah, I'm planning to clean up OnTapController next!

File: Sources/App/Slack/SlackListener.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+//
+//  Created by Allen Humphreys on 3/2/19.
+//
+
+import SlackKit
+import Vapor
+
+// Traditional singleton. Only a single listener is needed in the app
+// because the websocket client is single threaded
+final class SlackListener: ServiceType {
```
Comment (ichibod) [inline Sources/App/Slack/SlackListener.swift]: 👍

File: Sources/App/Slack/Slack.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            failure: { error in
+
+            }
+        )
+    }
+
+    func sendErrorMessage(text: String, channelId: String, user: String) throws {
+        guard let web = slackKit.webAPI else { throw Abort(.internalServerError) }
+
+        web.sendEphemeral(channel: channelId, text: text, user: user, success: { (ts, channel) in
```
Comment (ichibod) [inline Sources/App/Slack/Slack.swift]: Should/can we add thread?

File: Sources/App/Slack/Slack.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+//
+//  Created by Allen Humphreys on 3/2/19.
+//
+
+import SlackKit
+import Vapor
+
+struct Slack: ServiceType {
+    private let slackKit = SlackKit()
+    private let listener: SlackListener
```
Comment (ichibod) [inline Sources/App/Slack/Slack.swift]: I'm not sure how this all pieces together, but I'm worried about possible cyclical references with some of these

File: Sources/App/Slack/Slack.swift (hunk 3 above)
Comment (allenhumphreys) [reply Sources/App/Slack/Slack.swift]: The outgoing messages all use the Web API and there is an instance of `Slack` for each Vapor worker thread.

File: Sources/App/Slack/Slack.swift (hunk 4 above)
Comment (allenhumphreys) [reply Sources/App/Slack/Slack.swift]: cyclical references? Like retain cycles? Yes there is definitely circular references but all the instances are long lived

**Pull request merged**